File: game/forest/hurt_rock_animation.py
from animation import Animation
from forest.forest_resources import ForestResources
from forest.hurt_rock_hit_animation import HurtRockHitAnimation
from game_data import GameData
from render_type import RenderType
from state import State


class HurtRockAnimation(State):
    def __init__(self, context: GameData):
        super().__init__(context)
        self.needs_bottom = RenderType.POST

    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.hugo_lookrock):
            return HurtRockHitAnimation

        return None

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.hugo_lookrock, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtRockAnimation.ts
/** Port of game/forest/hurt_rock_animation.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { ForestResources } from './forestResources';
import { HurtRockHitAnimation } from './hurtRockHitAnimation';
import { RenderType } from '../renderType';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class HurtRockAnimation extends State {
  readonly needsBottom = RenderType.POST;

  constructor(context: GameData) { super(context); }

  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndexFast() >= ForestResources.hugoLookrock.length ? HurtRockHitAnimation : null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.hugoLookrock, this.getFrameIndexFast()), 0, 0);
  }
}
